clap_derive/src/derives/from_arg_matches.rs

```diff
@@ -126,108 +131,108 @@ pub fn gen_constructor(fields: &Punctuated<Field, Comma>, parent_attribute: &Att
                 let parser = attrs.parser();
                 let func = &parser.func;
                 let span = parser.kind.span();
-                let (value_of, values_of, mut parse) = match *parser.kind {
+
+                // The operand type of the `parse` function.
+                let parse_operand = match *parser.kind {
+                    FromStr | TryFromStr => quote_spanned!(ty.span()=> &str),
+                    FromOsStr | TryFromOsStr => quote_spanned!(ty.span()=> &::std::ffi::OsStr),
```

Comment:
Not sure why `parse_operand` is needed but we don't use `OsStr`. We use a crate by dylni.

Reply:
`os_str_bytes` is useful, but it still uses the `OsStr` type. For example, [`ArgMatches::value_of_os`](https://docs.rs/clap-v3/3.0.0-beta.1/clap_v3/struct.ArgMatches.html#method.value_of_os) returns an `Option<&OsStr>`.

It's possible the name `parse_operand` is misleading here. I've now renamed it to `parse_operand_type` since it's just a quote of the operand type, such as `&str` of the parse function.